app/controllers/profile/group_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Profile::GroupController < ApplicationController
+  def new
+    @groups = Group.all
+  end
+
+  def create
+    group_id = group_params[:group_select]
+    group = Group.find_by(id: group_id)
+    if group
```

Comment:
this check is made incase a user edits their html and try to join an invalid group or a group is deleted before they are able to join one - will add actual tests for these edge cases later